sklearn/model_selection/_search.py

```diff
@@ -557,54 +559,138 @@ def _fit(self, X, y, labels, parameter_iterable):
           for parameters in parameter_iterable
           for train, test in cv.split(X, y, labels))
 
-        # Out is a list of triplet: score, estimator, n_test_samples
-        n_fits = len(out)
+        all_test_scores, all_test_sample_counts, _, all_parameters = zip(*out)
+
+        # Get all the parameters
+        self._parameter_names = sorted(set(chain(*map(six.iterkeys,
+                                                      all_parameters))))
+
+        # Convert the results to numpy arrays
+        all_test_scores = np.array(all_test_scores, dtype=np.float64)
+        all_test_sample_counts = np.array(all_test_sample_counts, dtype=np.int)
+
+        n_splits = cv.get_n_splits(X, y, labels)
+        self.n_fits_ = all_test_scores.shape[0]
+        self.n_candidates_ = self.n_fits_ // n_splits
```

Comment:
If we need these to be stored on the object at all, surely it's because they're public. But I think n_candidates should be obvious from `results_`, while `n_fits_` is not obvious.